backend/app/schemas/slot.py

```
@@ -0,0 +1,66 @@
+    def validate_slot_window(cls, v: str) -> str:
+        """Validate that slot_window is one of the predefined time windows."""
+        if v not in SLOT_WINDOWS:
+            raise ValueError(f"slot_window must be one of {SLOT_WINDOWS}")
+        return v
+
+    @field_validator("slot_date")
+    @classmethod
+    def validate_not_past(cls, v: date) -> date:
+        """Validate that slot_date is not in the past."""
+        if v < datetime.now(timezone.utc).date():
+            raise ValueError("slot_date cannot be in the past")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)